closure/goog/net/jsloader.js

```diff
@@ -94,36 +94,26 @@ goog.net.jsloader.scriptsToLoad_ = [];
  * @param {Array<string>} uris The URIs to load.
  * @param {goog.net.jsloader.Options=} opt_options Optional parameters. See
  *     goog.net.jsloader.options documentation for details.
+ * @return {!goog.async.Deferred} The deferred result, that may be used to add
+ *     callbacks
  */
 goog.net.jsloader.loadMany = function(uris, opt_options) {
   // Loading the scripts in serial introduces asynchronosity into the flow.
-  // Therefore, there are race conditions where client A can kick off the load
-  // sequence for client B, even though client A's scripts haven't all been
-  // loaded yet.
-  //
-  // To work around this issue, all module loads share a queue.
+  // The client receives a deferred to manage parallel loads.
   if (!uris.length) {
-    return;
-  }
-
-  var isAnotherModuleLoading = goog.net.jsloader.scriptsToLoad_.length;
-  goog.array.extend(goog.net.jsloader.scriptsToLoad_, uris);
-  if (isAnotherModuleLoading) {
-    // jsloader is still loading some other scripts.
-    // In order to prevent the race condition noted above, we just add
-    // these URIs to the end of the scripts' queue and return.
-    return;
```

Comment:
I’ll do as you suggest. However, a quick request for comment:
It seems to me that the code loses a tad elegance.
To support the original use-case, we now need to store active deferred to return it in the if (isAnotherModuleLoading) clause.
But since we always return deferreds now, that use-case could easily be handled by the caller.
I'm happy to code for it in loadMany, but as "with queue" is more specific than w/o (judged by the number of lines), it would seem more natural to use the option to enable it ("..plus one uri queue please") rather than disable it ("..oh, without that queue thing please").
I am aware that backwards compatibility can be a issue somewhere.
With that, could you confirm that you prefer it this way and I'll PR per your spec.